crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -55,13 +58,18 @@ where
     /// See also [EthFilterConfig].
     ///
     /// This also spawns a task that periodically clears stale filters.
-    pub fn new(
+    pub fn new<Events, Network>(
         provider: Provider,
         pool: Pool,
         eth_cache: EthStateCache,
         config: EthFilterConfig,
         task_spawner: Box<dyn TaskSpawner>,
-    ) -> Self {
+        pubsub: EthPubSub<Provider, Pool, Events, Network>,
```

Comment:
we don't need this type, we only need events

Reply:
can you explain this further?

Reply:
we only need the events trait not the pubsub isntance